tuf/repository_lib.py

```diff
@@ -1253,20 +1264,25 @@ def generate_targets_metadata(targets_directory, target_files, version,
   filedict = {}
 
   if use_existing_fileinfo:
+    # Use the provided fileinfo dicts, conforming to FILEINFO_SCHEMA, rather than
+    # generating fileinfo
     for target, fileinfo in six.iteritems(target_files):
 
       # Ensure all fileinfo entries in target_files have a non-empty hashes dict
       if not fileinfo.get('hashes', None):
         raise securesystemslib.exceptions.Error('use_existing_hashes option set'
             ' but no hashes exist in roledb for ' + repr(target))
 
-      if fileinfo.get('length', -1) < 0:
+      # and a non-empty, non-zero, length
+      if fileinfo.get('length', -1) < 1:
```

Comment:
I think it doesn't make sense to have zero length targets files, but maybe this should be restored to the prior check of: `if len < 0` ?

Reply:
> I think it doesn't make sense to have zero length targets files, but maybe this should be restored to the prior check of: `if len < 0` ?

@joshuagl, the classical `__init__.py` file is a good example for a target file with zero length.

Reply:
You're right, thanks for pointing out that case @lukpueh. In which case I should restore this to the prior `if len < 0`. Will force push a fix later today.

Reply:
Oh, I already restored the behaviour for checking `len < 0` in #1034, which contains an updated version of this change. Apologies for the confusion.

Reply:
No need to apologize. I have no business commenting on closed PRs. :D